internal/pkg/service/cli/dialog/create_table.go

```diff
@@ -28,49 +40,106 @@ func (p *Dialogs) AskCreateTable(args []string, branchKey keboola.BranchKey, all
+	columnsMethod, err := p.columnsDefinitionMethod()
+	if err != nil {
+		return opts, err
+	}
+
+	switch columnsMethod {
+	case columnsNamesFlag:
+		columnsStr := p.options.GetString(`columns`)
+		colNames := strings.Split(strings.TrimSpace(columnsStr), ",")
+		opts.CreateTableRequest.Columns = getOptionCreateRequest(colNames)
+	case columnsDefinitionFlag:
+		filePath := p.options.GetString("columns-from")
+		columnsDefinition, err := parseJSONInputForCreateTable(filePath)
+		if err != nil {
+			return table.Options{}, err
+		}
+		opts.CreateTableRequest.Columns = columnsDefinition
+	case columnsNamesInteractive:
+		columnsStr := p.options.GetString(`columns`)
+		if !p.options.IsSet(`columns`) {
+			columnsStr, _ = p.Ask(&prompt.Question{
+				Label:       "Columns",
+				Description: "Enter a comma-separated list of column names.",
+			})
+		}
+		colNames := strings.Split(strings.TrimSpace(columnsStr), ",")
+		opts.CreateTableRequest.Columns = getOptionCreateRequest(colNames)
+	case columnsDefinitionInteractive:
+		input, _ := p.Editor("yaml", &prompt.Question{
+			Label:       "Columns definition from file",
+			Description: "Columns definition from file",
+			Default:     p.defaultValue(),
+			Validator: func(val any) error {
+				_, err := parseColumnsDefinitionFromFile(val.(string))
+				if err != nil {
+					return err
+				}
+				return nil
+			},
 		})
+		res, err := parseColumnsDefinitionFromFile(input)
+		if err != nil {
+			return table.Options{}, err
+		}
+		opts.CreateTableRequest.Columns = res
+	default:
+		panic(errors.New("unexpected state"))
 	}
-	colNames := strings.Split(strings.TrimSpace(columnsStr), ",")
 
+	// Primary keys
 	if p.options.IsSet(`primary-key`) {
 		opts.CreateTableRequest.PrimaryKeyNames = strings.Split(strings.TrimSpace(p.options.GetString(`primary-key`)), ",")
 	} else {
 		primaryKey, _ := p.MultiSelect(&prompt.MultiSelect{
 			Label:   "Select columns for primary key",
-			Options: colNames,
+			Options: getColumnsName(opts.CreateTableRequest.Columns),
 		})
 		opts.CreateTableRequest.PrimaryKeyNames = primaryKey
 	}
 
-	filePath := p.options.GetString("columns-from")
-	if p.options.IsSet("columns-from") {
-		columnsDefinition, err := parseJSONInputForCreateTable(filePath)
-		if err != nil {
-			return table.Options{}, err
-		}
+	return opts, nil
+}
 
-		opts.CreateTableRequest.Columns = columnsDefinition
-	} else {
-		opts.CreateTableRequest.Columns = getOptionCreateRequest(strings.Split(strings.TrimSpace(columnsStr), ","))
+func (p *Dialogs) columnsDefinitionMethod() (columnsDefinitionMethod, error) {
+	switch {
+	case !p.options.IsSet("columns-from") && !p.options.IsSet("columns"):
+		// Ask for method
+		specifyTypes := p.Prompt.Confirm(&prompt.Confirm{
+			Label:       "Columns Types Definition",
+			Description: "Would you define column types?",
+			Default:     true,
+			Help:        `"In Keboola you can define column types, if types won't be specified default value will be "string"`,
+		})
```

Comment:
Maybe it would be simpler:

Label: `Column types`
Description: `Do you want to specify column types? Otherwise all columns default to strings.`
Help: n/a